### PR TITLE
Chartboost - Release adapters 8.0.1.0

### PR DESCRIPTION
**Pull request by stosiarz**
- Updated Chartboost sdk to version 8.0.1
- Integration changes for adapers
- Implements new banner adapter

### CODE REVIEW

File: Chartboost/src/main/java/com/mopub/mobileads/ChartboostAdapterConfiguration.java [hunk 1]
```diff
@@ -76,9 +75,9 @@ public void initializeNetwork(@NonNull Context context, @Nullable Map<String, St
 
         synchronized (ChartboostAdapterConfiguration.class) {
             try {
-                if (configuration != null && context instanceof Activity) {
+                if (configuration != null) {
```
Comment (hmertkahraman) [inline Chartboost/src/main/java/com/mopub/mobileads/ChartboostAdapterConfiguration.java]: Can add ` && !configuration.isEmpty()`

File: Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                    MoPubErrorCode.NETWORK_NO_FILL.getIntCode(),
+                    MoPubErrorCode.NETWORK_NO_FILL);
+            return;
+        }
+
+
+        final BannerSize bannerSize = chartboostAdSizeFromLocalExtras(localExtras);
+
+        internalView = new FrameLayout(context);
+        FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.WRAP_CONTENT);
```
[the remark this answers is not in the document]
Comment (stosiarz) [reply Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java]: SDK is available and can be tested. I tested it with mopub sandbox application and banner was displayed as expected.

File: Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        lp.gravity = Gravity.CENTER_HORIZONTAL;
+        internalView.setLayoutParams(lp);
+
+        banner = new ChartboostBanner(context, mLocation, bannerSize, bannerListener);
+        banner.removeAllViews();
+        internalView.addView(banner);
+        mCustomEventBannerListener.onBannerLoaded(internalView);
+        banner.show();
```
[the remark this answers is not in the document]
Comment (stosiarz) [reply Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java]: Show is required to display the cached banner. In case banner was not cached previously, show will cache it automatically and will display the banner.

File: Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            if(chartboostCacheError!=null) {
+                MoPubLog.log(getAdNetworkId(), LOAD_FAILED, ADAPTER_NAME,
+                        chartboostCacheError.code,
+                        chartboostCacheError.toString());
+                mCustomEventBannerListener.onBannerFailed(MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
+            }
+        }
+
+        @Override
+        public void onAdShown(ChartboostShowEvent chartboostShowEvent, ChartboostShowError chartboostShowError) {
```
[the remark this answers is not in the document]
Comment (stosiarz) [reply Chartboost/src/main/java/com/mopub/mobileads/ChartboostBannerCustomEvent.java]: impression is done automatically inside Chartboost sdk

**Comment by hmertkahraman** [comment]
The file under libs/mopub-chartboost-adapters-7.3.0.0.jar can be removed.

File: Chartboost/src/main/java/com/mopub/mobileads/ChartboostRewardedVideo.java [hunk 5]
```diff
@@ -158,22 +158,18 @@ protected void onInvalidate() {
     private static final class ChartboostLifecycleListener implements LifecycleListener {
         @Override
         public void onCreate(@NonNull Activity activity) {
-            Chartboost.onCreate(activity);
```
Comment (hmertkahraman) [inline Chartboost/src/main/java/com/mopub/mobileads/ChartboostRewardedVideo.java]: @stosiarz May I ask why did you remove the lifecycle listener events mapping for this and the following methods? Is it linked with the move to Context from Activity? Asking this because ideally networks should be matching the lifecycle events with their own especially for control on fullscreen ads, [example here](https://developers.mopub.com/publishers/android/interstitial/#best-practices).

Comment (stosiarz) [reply Chartboost/src/main/java/com/mopub/mobileads/ChartboostRewardedVideo.java]: We removed activity dependency from Chartboost and all related lifecycle events from public API. onCreate and other related methods don't exist in Chartboost class anymore.
